ssc_interface_wrapper_ros2/src/ssc_interface_wrapper_node.cpp

```diff
@@ -0,0 +1,174 @@
+        robot_status_pub_   = create_publisher<carma_driver_msgs::msg::RobotEnabled>("controller/robot_status", 1);
+        vehicle_engage_pub_ = create_publisher<std_msgs::msg::Bool>("vehicle/engage", 5);                                                                                                                                                                                        
+        
+        // initialize services
+        enable_robotic_control_srv_ = create_service<carma_driver_msgs::srv::SetEnableRobotic>("controller/enable_robotic", 
+                                                                                    std::bind(&Node::enable_robotic_control_cb, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));                                                                  
+
+
+    }
+
+    carma_ros2_utils::CallbackReturn Node::handle_on_activate(const rclcpp_lifecycle::State& )
+    {
+        //Timer setup
+        timer_ = this->create_wall_timer(std::chrono::milliseconds(500), 
+        std::bind(&Node::check_driver_timeout, this));
+
+        if(reengage_state_)
+        {
+            reengage_robotic_control();
+        }
+
+    }
+
+    void Node::check_driver_timeout()
+    {
+        worker_.check_driver_status(this->now(), config_.controller_timeout);
+    }
+
+    bool Node::enable_robotic_control_cb(const std::shared_ptr<rmw_request_id_t> header,
+                                        const std::shared_ptr<carma_driver_msgs::srv::SetEnableRobotic::Request> req, 
+                                        const std::shared_ptr<carma_driver_msgs::srv::SetEnableRobotic::Response> resp)
+    {
+        std_msgs::msg::Bool engage_cmd;
+        std::string current_state = worker_.get_current_ssc_state();
+        
+        if(!current_state.compare("ready")) {
+            if(req->set == carma_driver_msgs::srv::SetEnableRobotic::Request::ENABLE) {
+                engage_cmd.data = true;
+                vehicle_engage_pub_->publish(engage_cmd);
+            }
+        } else if(!current_state.compare("active")) {
+            if(req->set == carma_driver_msgs::srv::SetEnableRobotic::Request::DISABLE) {
+                engage_cmd.data = false;
+                vehicle_engage_pub_->publish(engage_cmd);
+            }
+        } else if(!current_state.compare("failure")) {
+            engage_cmd.data = false;
+            vehicle_engage_pub_->publish(engage_cmd);
+            if(req->set == carma_driver_msgs::srv::SetEnableRobotic::Request::ENABLE) {
+                reengage_state_ = true;
+            }
+        }
+        return true;
+
+    }
+
+    void Node::ssc_state_cb(const automotive_navigation_msgs::msg::ModuleState::UniquePtr msg)
+    {
+        // message needs to go to library. Unique ptr cant be sent
+        automotive_navigation_msgs::msg::ModuleState module_state = *msg.get();
```

Comment:
Not really sure what *msg.get() would give
Just *msg should be sufficient to access the value pointed to.